diff --git a/m4/check-libc.m4 b/m4/check-libc.m4
index 3e4d3af..ab42f77 100644
--- a/m4/check-libc.m4
+++ b/m4/check-libc.m4
@@ -2,10 +2,26 @@ AC_DEFUN([CHECK_LIBC_COMPAT], [
 # Check for libc headers
 AC_CHECK_HEADERS([err.h readpassphrase.h])
 # Check for general libc functions
-AC_CHECK_FUNCS([asprintf getpagesize inet_ntop inet_pton memmem readpassphrase])
+AC_CHECK_FUNCS([asprintf inet_ntop inet_pton memmem readpassphrase])
 AC_CHECK_FUNCS([reallocarray recallocarray])
 AC_CHECK_FUNCS([strlcat strlcpy strndup strnlen strsep strtonum])
 AC_CHECK_FUNCS([timegm _mkgmtime])
+
+dnl Note: in Android <= L getpagesize() is defined inline. As such, the test
+dnl for getpagesize() should include <unistd.h>.
+AC_MSG_CHECKING([for getpagesize])
+AC_LINK_IFELSE([AC_LANG_PROGRAM([
+        #include <unistd.h>
+], [
+        return getpagesize();
+])], [
+  ac_cv_func_getpagesize="yes"
+  AC_MSG_RESULT([yes])
+], [
+  ac_cv_func_getpagesize="no"
+  AC_MSG_RESULT([no])
+])
+
 AM_CONDITIONAL([HAVE_ASPRINTF], [test "x$ac_cv_func_asprintf" = xyes])
 AM_CONDITIONAL([HAVE_GETPAGESIZE], [test "x$ac_cv_func_getpagesize" = xyes])
 AM_CONDITIONAL([HAVE_INET_NTOP], [test "x$ac_cv_func_inet_ntop" = xyes])
