diff --git a/Makefile.am b/Makefile.am
index 0f741bc..3a3eee0 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -91,10 +91,6 @@ VERSION_INFO = 6:2:0
 # is user-visible, and so we can pretty much guarantee that release
 # series won't be binary-compatible.
 
-if INSTALL_LIBEVENT
-dist_bin_SCRIPTS = event_rpcgen.py
-endif
-
 pkgconfigdir=$(libdir)/pkgconfig
 LIBEVENT_PKGCONFIG=libevent.pc libevent_core.pc libevent_extra.pc
 
@@ -130,7 +126,6 @@ endif
 
 if INSTALL_LIBEVENT
 lib_LTLIBRARIES = $(LIBEVENT_LIBS_LA)
-pkgconfig_DATA = $(LIBEVENT_PKGCONFIG)
 else
 noinst_LTLIBRARIES =  $(LIBEVENT_LIBS_LA)
 endif
@@ -143,8 +138,6 @@ CLEANFILES=
 DISTCLEANFILES=
 BUILT_SOURCES =
 include include/include.am
-include sample/include.am
-include test/include.am
 
 if BUILD_WIN32
 
diff --git a/configure.ac b/configure.ac
index 7528d37..24b65dc 100644
--- a/configure.ac
+++ b/configure.ac
@@ -121,10 +121,6 @@ AC_ARG_ENABLE([function-sections],
 AC_ARG_ENABLE([verbose-debug],
 		AS_HELP_STRING([--enable-verbose-debug, verbose debug logging]),
 	[], [enable_verbose_debug=no])
-AC_ARG_ENABLE([clock-gettime],
-     AS_HELP_STRING(--disable-clock-gettime, do not use clock_gettime even if it is available),
-  [], [enable_clock_gettime=yes])
-
 
 AC_PROG_LIBTOOL
 
@@ -141,10 +137,6 @@ dnl Checks for libraries.
 AC_SEARCH_LIBS([inet_ntoa], [nsl])
 AC_SEARCH_LIBS([socket], [socket])
 AC_SEARCH_LIBS([inet_aton], [resolv])
-if test "x$enable_clock_gettime" = "xyes"; then
-  AC_SEARCH_LIBS([clock_gettime], [rt])
-  AC_CHECK_FUNCS([clock_gettime])
-fi
 AC_SEARCH_LIBS([sendfile], [sendfile])
 
 dnl - check if the macro _WIN32 is defined on this compiler.
