diff --git a/CMakeLists.txt b/CMakeLists.txt
index b4a34f3..27fa4f7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1430,6 +1432,8 @@ set(LIBEVENT_LIBRARIES
         event_extra
         CACHE STRING "Libevent libraries")
 
+string(REPLACE "/MD" "/MT" CMAKE_C_FLAGS_RELEASE ${CMAKE_C_FLAGS_RELEASE})
+
 message(STATUS "")
 message(STATUS "        ---( Libevent " ${EVENT_VERSION} " )---")
 message(STATUS "")
